lib/worker/OperationQueue.js

```diff
@@ -81,19 +83,18 @@ module.exports = class OperationQueue {
       });
     }
 
-    // if the queue is not initialized or the head of the queue is at the end
-    // of the queue, then no next chunk of operations cached yet, so create one
-    // that will be cached when `getNextChunk` is called and its operations are
-    // retrieved
-    if(!(Array.isArray(this.queue) && (this.head + 1) < this.queue.length)) {
+    // if the queue is not initialized with the head of the queue before the end
+    // of the queue, then no next chunk of operations is cached yet, so create
+    // one that will be cached when `getNextChunk` is called and its operations
+    // are retrieved
+    const end = (this.queue || []).length - 1;
```

Comment:
We can avoid the unnecessary array creation:

```suggestion
    const end = this.queue ? this.queue.length - 1 : -1;
```

And did you mean for `end` to be `-1` or `0` in the case of no queue?

Reply:
Is the array not lazily created here? If the intent was to prevent allocating a new array entirely, your suggestion is a solution.

Reply:
Yes, that was my intent.